### PR TITLE
Fix several Kconfig leaks in device drivers.

### PR DESCRIPTION
**Pull request by kartben**
Just did a pass across drivers and spotted some leakages. See individual commits.

### CODE REVIEW

**Review by decsny — APPROVED** [review]

**Comment by bjarki-trackunit** [comment]
Is there a reason to use
```
config FOO
        bool "some foo"

config FOO_BAR
        int "some foo bar"
        depends on FOO
```
rather than
```
config FOO
        bool "some foo"

if FOO

config FOO_BAR
        int "some foo bar"

endif # FOO
```
other than its how the other configs are written?

**Review by tmon-nordic — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

File: drivers/regulator/Kconfig.nxp_vref [hunk 1]
```diff
@@ -3,12 +3,14 @@
 
 config REGULATOR_NXP_VREF
 	bool "NXP VREF peripheral driver"
-	default y if DT_HAS_NXP_VREF_ENABLED
+	default y
```
Comment (dleach02) [inline drivers/regulator/Kconfig.nxp_vref:6]: Why this change?

File: drivers/timer/Kconfig.nrf_grtc [hunk 2]
```diff
@@ -3,7 +3,8 @@
 
 menuconfig NRF_GRTC_TIMER
 	bool "nRF GRTC Timer"
-	default y if DT_HAS_NORDIC_NRF_GRTC_ENABLED
+	default y
+	depends on DT_HAS_NORDIC_NRF_GRTC_ENABLED
```
[the remark this answers is not in the document]
Comment (kartben) [reply drivers/timer/Kconfig.nrf_grtc:7]: Without the "depends on" it shows up unconditionally and clutters the UI as well as ends up in the final config file

File: drivers/regulator/Kconfig.nxp_vref (hunk 1 above)
Comment (kartben) [reply drivers/regulator/Kconfig.nxp_vref:6]: Ditto

**Review by bjarki-andreasen — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]
